File: 8-kyu-kata/function-1-hello-world/javascript/function1HelloWorld.Test.js
// 8 kyu - Function 1 - hello world  [ ID: 523b4ff7adca849afe000035  (function-1-hello-world) ]
// URL: https://www.codewars.com/kata/523b4ff7adca849afe000035
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
const assert = require("chai").assert

describe("Testing function", function () {
  it("Is it a function?", function () {
    assert.strictEqual(typeof greet, "function", "greet should be a function")
  })
  it("Correct return-value?", function () {
    assert.strictEqual(greet(), "hello world!")
  })
})

File: kata-8-kyu/function-1-hello-world/python/function_1_hello_world_test.py
# + =====================================================================================================================
# +
# + 8 kyu - Function 1 - hello world  [ ID: 523b4ff7adca849afe000035 ] (function-1-hello-world)
# + URL: https://www.codewars.com/kata/523b4ff7adca849afe000035
# + Category: REFERENCE  |  Tags: FUNDAMENTALS
# +
# + =====================================================================================================================

import codewars_test as test
from function_1_hello_world import greet


@test.describe("Greet function")
def _():
    @test.it("Making sure greet exists")
    def _():
        try:
            test.expect(greet)
        except NameError:
            test.fail("Greet doesn't exist")

    @test.it("Testing that it returns hello world!")
    def _():
        test.assert_equals(greet(), "hello world!", "Greet doesn't return hello world!")
